apps/gitness/src/pages/repo/repo-files.tsx

```diff
@@ -21,24 +120,41 @@ export const RepoFiles: React.FC = () => {
   const navigate = useNavigate()
 
   const [selectedBranch, setSelectedBranch] = useState<string>(gitRef || '')
+  const [isOpen, setIsOpen] = useState(false)
+  const [query, setQuery] = useState('')
 
   const { data: repository } = useFindRepositoryQuery({ repo_ref: repoRef })
 
   const { data: branches } = useListBranchesQuery({
     repo_ref: repoRef,
-    queryParams: { include_commit: false, sort: 'date', order: 'asc', limit: 20, page: 1, query: '' }
+    queryParams: {
+      include_commit: false,
+      sort: 'date',
+      order: 'asc',
+      limit: 20,
+      page: 1,
+      query: ''
```

Comment:
when you add query to this, probably add a debounce so it doesnt rerender multiple times on input change

Reply:
we are not changing this query anywhere as of now